frontend/packages/kubevirt-plugin/src/components/modals/vm-boot-order-modal/boot-order.tsx

```diff
@@ -0,0 +1,175 @@
+];
+
+export const BootOrderModal = withHandlePromise((props: BootOrderModalProps) => {
+  const { vmLikeEntity, handlePromise, onClose } = props;
+  const [state, setState] = React.useState<BootOrderModalState>({
+    sources: getBootableDevices(vmLikeEntity),
+    options: getUnBootableDevices(vmLikeEntity),
+    isEditMode: false,
+    changed: false,
+  });
+
+  const setEditMode = (mode: boolean): void => {
+    setState({
+      sources: state.sources,
+      options: state.options,
+      isEditMode: mode,
+      changed: state.changed,
+    });
+  };
+
+  // Move an item form the options list to the sources list, item key is "<type>->name>".
+  const addByKey = (key: string): void => {
+    const item = _.find(state.options, (option) => `${option.type}-${option.value.name}` === key);
+    const options = _.filter(
+      state.options,
+      (option) => `${option.type}-${option.value.name}` !== key,
+    );
+
+    const sources = [...state.sources, item];
+    relaxSources(sources);
+
+    setState({ sources, options, isEditMode: false, changed: true });
+  };
+
+  const onDelete = (index: number) => {
+    const item = state.sources[index];
+    delete item.value.bootOrder;
+
+    const sources = [...state.sources.slice(0, index), ...state.sources.slice(index + 1)];
+    relaxSources(sources);
+    const options = [...state.options, item];
+
+    setState({ sources, options, isEditMode: false, changed: true });
+  };
+  const onMove = (index: number, toIndex: number) => {
+    const unMovedSources = [...state.sources.slice(0, index), ...state.sources.slice(index + 1)];
+    const sources = [
+      ...unMovedSources.slice(0, toIndex),
+      state.sources[index],
+      ...unMovedSources.slice(toIndex),
+    ];
+    relaxSources(sources);
+
+    setState({ sources, options: state.options, isEditMode: false, changed: true });
+  };
+
+  const submit = async (e) => {
+    e.preventDefault();
+    if (state.changed) {
+      const patches = [
+        new PatchBuilder('/spec/template/spec/domain/devices/disks')
+          .setOperation(PatchOperation.REPLACE)
+          .setValue(getDisks(state.sources, state.options))
+          .build(),
+        new PatchBuilder('/spec/template/spec/domain/devices/interfaces')
+          .setOperation(PatchOperation.REPLACE)
+          .setValue(getInterfaces(state.sources, state.options))
+          .build(),
+      ];
+      const promise = k8sPatch(getVMLikeModel(vmLikeEntity), vmLikeEntity, patches);
+      handlePromise(promise).then(onClose); // eslint-disable-line promise/catch-or-return
+    } else {
+      onClose();
+    }
+  };
+
+  const itemRow: React.FC<{ index: number }> = ({ index }) => {
+    const item = state.sources[index];
+    return <p>{`${item.value.name} (${item.typeLabel})`}</p>;
+  };
+  const addItemRow: React.FC<any> = AddItemRowFactory(state.options, addByKey);
+
+  return (
+    <Modal
```

Comment:
Could you please extract the modal part to a separate component (due to the reusabilty for the VM wizard)?

Something like a `disk-modal.tsx` and `disk-modal-enhanced.tsx`. The difference is that boot order will not be represented as a modal when inside the VM wizard. It will be a second advanced tab (likely a followup PR to this one).